### DISCUSSION TITLE
ObservableObject in WPF

### DISCUSSION THREAD

**Discussion by malsabi** [CommunityToolkit/dotnet#105] [Q&A]
I'm using MVVM Toolkit in a WPF Project, I have a View-Model 'DashboardViewModel' that inherits ObservableObject. The issue is that when I set the Data-Context in the xaml view I'm getting these exceptions:

![Exceptions](https://user-images.githubusercontent.com/21250940/151651010-01851946-5793-4c13-b10a-737e03f1afd7.png)

- If I removed the ObservableObject from the View-Model ('DashboardViewModel') it will work fine as expected.

Also here is how I'm setting the Data-Context in the xaml file:

![ViewModelException](https://user-images.githubusercontent.com/21250940/151651062-4a9d7d98-6ecb-4e0d-bd0f-4f1734729f7e.png)

NOTES::
1. This approach was done in UWP and I had no errors/exceptions.
2. I can fix the issue by setting the View-Model in the Code-Behind as the following:
![ViewModelException2](https://user-images.githubusercontent.com/21250940/151651138-af50a0b7-63c7-474d-935e-e377ec3ba8ef.png)
Also there won't be any errors.
3. I can fix the issue by assigning a value to the d:DataContext that sets the design-time Data Context as the following:
![ViewModelException3](https://user-images.githubusercontent.com/21250940/151651384-ec31cb28-dee1-4f2f-93df-22e1eda304fc.png)
4. I can fix the issue by using View-Model-Locator technique.

Yet I prefer to use the first simple solution, Is there any work around for it ? I'm confused why it works in UWP and not in WPF, I guess maybe because the library is focusing more on the UWP.

**Comment by VacuumBreather** [comment]
Why do you need the x:Name attached property on there? From your problem description it seems that's the only problem?
